### DISCUSSION TITLE
What technologies does the kitty terminal use?

### DISCUSSION THREAD

**Discussion by dclong** [kovidgoyal/kitty#3561] [Q&A]
I see mixed usage of C, Python and GLSL. In particular, I'm interested in the role of Python script in the project.

**Comment by kovidgoyal** [comment]
C is used for performance sensitive code and Python for UI/scripting.

**Reply by dclong** [reply]
May I ask which UI framework is used?

**Comment by kovidgoyal** [comment]
On Sat, May 01, 2021 at 11:49:02PM -0700, Ben Chuanlong Du wrote:
> May I ask which UI framework is used?

None. All drawing is done directly via OpenGL. Window system interaction
is handled by a much improved private fork of GLFW.
